tests/devices/test_default_qutrit_mixed.py

```diff
@@ -1270,3 +1270,305 @@ def qfunc(x, y):
+        qml.QutritUnitary(self.setup_unitary, wires=0)
+        qml.QutritUnitary(self.setup_unitary, wires=1)
+        if nr_wires == 3:
+            qml.TAdd(wires=(0, 2))
+
+    @staticmethod
+    def get_expected_dm(num_wires):
+        """Gets the expected density matrix of the circuit for the first num_wires"""
+        state = np.array([2, 3, 6], dtype=complex) ** -(1 / 2)
+        if num_wires == 2:
+            state = np.kron(state, state)
+        if num_wires == 3:
+            state = sum(
+                [
+                    state[i] * reduce(np.kron, [v, state, v])
+                    for i, v in enumerate([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+                ]
+            )
+        return np.outer(state, state)
+
+    # Set up the gammas and probs that are inputted as the measurement error.
+    gammas_and_probs = [
+        [(0, 0, 0), (0, 0, 0)],
+        [None, (1, 0, 0)],
+        [None, (0, 1, 0)],
+        [None, (0, 0, 1)],
+        [(1, 0, 0), None],
+        [(0, 1, 0), None],
+        [(1, 0, 1), None],
+        [(0, 1, 0), (0, 0, 1)],
+        [None, (0.1, 0.2, 0.4)],
+        [(0.2, 0.1, 0.3), None],
+        [(0.2, 0.1, 0.4), (0.1, 0.2, 0.5)],
+    ]
+
+    # Expected probabilities of measuring each state after the above measurement errors are applied.
+    expected_probs = [
+        [1 / 2, 1 / 3, 1 / 6],
+        [1 / 3, 1 / 2, 1 / 6],
+        [1 / 6, 1 / 3, 1 / 2],
+        [1 / 2, 1 / 6, 1 / 3],
+        [5 / 6, 0, 1 / 6],
+        [2 / 3, 1 / 3, 0],
+        [5 / 6, 1 / 6, 0],
+        [2 / 3, 0, 1 / 3],
+        [5 / 12, 17 / 60, 0.3],
+        [7 / 12, 19 / 60, 0.1],
+        [11 / 24, 7 / 30, 37 / 120],
+    ]
+
+    @pytest.mark.parametrize("gammas_and_probs, expected", zip(gammas_and_probs, expected_probs))
+    def test_readout_probs(self, nr_wires, gammas_and_probs, expected):
+        """Tests the measurement results for probs"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            damping_measurement_gammas=gammas_and_probs[0],
+            trit_flip_measurement_probs=gammas_and_probs[1],
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return qml.probs(wires=0)
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    # Expected expval list from circuit with diagonal observables after the measurement errors
+    # defined by gammas_and_probs are applied.
+    expected_commuting_expvals = [
+        [1 / 6, 1 / (2 * np.sqrt(3)), 1 / 6],
+        [-1 / 6, 1 / (2 * np.sqrt(3)), -1 / 6],
+        [-1 / 6, -1 / (2 * np.sqrt(3)), -1 / 6],
+        [1 / 3, 0, 1 / 3],
+        [5 / 6, 1 / (2 * np.sqrt(3)), 5 / 6],
+        [1 / 3, 1 / np.sqrt(3), 1 / 3],
+        [2 / 3, 1 / np.sqrt(3), 2 / 3],
+        [4 / 6, 0, 4 / 6],
+        [2 / 15, 1 / (10 * np.sqrt(3)), 2 / 15],
+        [4 / 15, 7 / (10 * np.sqrt(3)), 4 / 15],
+        [9 / 40, 3 / (40 * np.sqrt(3)), 9 / 40],
```

Comment:
praise: good job working all these out... this looks tedious :cry:

Reply:
Thank you, I don't want to admit how many times I had to redo it because I wanted to simplify or realized there was an issue with the non-commuting test.